apps/linkedin-to-notion/src/components/LinkedInNotionSidePanel/Form.tsx

```diff
@@ -77,19 +76,53 @@ export const Form = ({
       }
     };
 
-    setLinkedinUrl(window.location.href.match(/https:\/\/[a-z]{2,4}\.linkedin\.com\/in\/[^/]+\//gim)[0]);
-
     setFormValues();
   }, [displayNotionValues, linkedinValues]);
 
+  const findProfileInNotionDatabase = async (databaseId: string, linkedinUrl: string): Promise<void> => {
+    const res = await sendToBackground<
+      { notionToken: string; databaseId: string; linkedinUrl: string },
+      NotionProfileInformation
+    >({
+      name: 'notion/resolvers/findProfileInDatabase',
+      body: {
+        notionToken: notionToken.accessToken,
+        databaseId,
+        linkedinUrl,
+      },
+    });
+
+    if (res) {
+      if ((res as unknown as ErrorResponse).error) {
+        console.log("The profile query didn't work", res);
+        setAlertState('error');
+        setCurrentNotionValues(null);
+        return;
+      } else {
+        // If we did find a profile
+        console.log('Profile found in Notion', res);
+        setCurrentNotionValues(res);
+        setNotionId(res.notionId);
+        setAlertState('in-notion');
+        return;
+      }
+    } else {
+      // The search worked but no profile in selected db
+      console.log('No profile found in Notion');
+      setCurrentNotionValues(null);
+      setAlertState('new-profile');
+      return;
+    }
```

Comment:
Same comments on else 😬


```suggestion
    if (!res) {
        console.log('No profile found in Notion');
        setCurrentNotionValues(null);
        setAlertState('new-profile');
        return;
    }

    if ((res as unknown as ErrorResponse).error) {
        console.log("The profile query didn't work", res);
        setAlertState('error');
        setCurrentNotionValues(null);
        return;
    }

    console.log('Profile found in Notion', res);
    setCurrentNotionValues(res);
    setNotionId(res.notionId);
    setAlertState('in-notion');
    return;
```